### DISCUSSION TITLE
Slow response

### DISCUSSION THREAD

**Discussion by b00tkitism** [coredns/coredns#5898] [Q&A]
Why CoreDns is so slow for first request for every domain ?

Edit: not first query, it become slow randomly

**Comment by chrisohaver** [comment]
environment? configuration? logs? errors?

**Comment by b00tkitism** [comment]
Ubuntu 18.04
Coredns 1.10
Corefile:
`.:53 {
    cache {
        prefetch 2 1s 20%
    }
    log
    errors
    reload
    cancel 1s
    file /root/db.any
    forward . X.X.X.X:53
}`

db file is a file with 2000-2500 records
Delay with dig is between 800ms-3000ms

**Reply by chrisohaver** [reply]
The random delays are probably due to sporadic network latency to the upstream server.

**Reply by b00tkitism** [reply]
No it wasn't, when the number of queries goes up to 6k per second it take so much delay

**Reply by chrisohaver** [reply]
Delay related to QPS could be due to resource constraints.

**Reply by b00tkitism** [reply]
8 Core cpu + 4 GB Ram
Bind9 works fine with same configurations.

**Reply by b00tkitism** [reply]
Coredns has 800% cpu usage while bind has 200%, why ?

**Comment by johnbelamaric** [comment]
We really need way, way more information to make this a productive conversation.

Logs, information about the queries being run and how you are testing it, information about the networking setup, the upstream domains, more detailed configuration, etc.

There's not enough information here to make any kind of reasonable guess.

**Reply by b00tkitism** [reply]
Ok, I'll share more information tomorrow

**Reply by johnbelamaric** [reply]
Thank you!
